### PR TITLE
Add unit test

### PR DESCRIPTION
**Pull request by manq2010**
OOP school library: unit tests
---

In this PR, the following was done:

- [x] Added unit tests for methods in the following classes
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- And other  classes implemented on the decorator design

### CODE REVIEW

File: spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'rspec'
+
+describe Classroom do
+  let(:classroom) { Classroom.new('Class 1') }
+  let(:student) { double('Student') }
+
+  it 'has a label' do
+    expect(classroom.label).to eq('Class 1')
+  end
```
Comment (Alfredbis29) [inline spec/classroom_spec.rb]: - [ ] Great work thus far :clap: However, all methods in all classes should have unit tests, therefore please make sure the `add_student` `classroom` function has (at least) a 'it' block in accordance with the project's specifications. check your file classroom.rb
 
Screenshot from 2023-02-10 16-42-18

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    it 'inherits the name attribute from Person' do
+      expect(@student).to have_attributes(age: 17)
+      expect(@student).to have_attributes(name: 'Young Tiger')
+      expect(@student).to have_attributes(parent_permission: true)
+    end
+
+    it 'has an empty list of teachers by default' do
+      expect(@student.rentals).to eq([])
+    end
```
Comment (Alfredbis29) [inline spec/student_spec.rb:21]: - [ ] Please be careful to provide unit tests for all of the following methods in the `student class`: The tests using the `classroom` technique are being missed by Cureent. :wink:  
Check this :point_down: 
Screenshot from 2023-02-10 16-58-28

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  #   describe '#to_json' do
+  #     it 'returns a JSON representation of the Student object' do
+  #       @student = Student.new('Jane Doe')
+  #       @student.teachers = ['John Smith', 'Jane Smith']
+  #       expect(@student.to_json).to eq({ 'name' => 'Jane Doe', 'teachers' => ['John Smith', 'Jane Smith'] })
+  #     end
+  #   end
```
Comment (Alfredbis29) [inline spec/student_spec.rb]: - [ Optional] KIndly remove unused codes in your project it make unprofessional :wink:

File: spec/teacher_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    end
+  end
+
+  #   describe '#to_json' do
+  #     it 'returns a JSON representation of the Teacher object' do
+  #       @person = Teacher.new(45, 'Jimmy Holder', 'Bio')
+  #       expect(@person.to_json).to eq({ 'age' => 45, 'name' => 'Jimmy Holder', 'class' => 'Teacher', 'rentals' => [],
+  #                                         'specialization' => 'Bio' })
+  #     end
+  #   end
```
Comment (Alfredbis29) [inline spec/teacher_spec.rb]: - [ ] Kindly remove unused codes in your project , it make your codes look ugly :wink:

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
### Hi Team,
 You did a great job so far 👏🏼 👏🏼 There are some issues that you still need to work on to go to the next project but you are almost there!
 
 
 ### To highlight 💪🏻
 * All test files is placed in a different directory. :heavy_check_mark:
* Gitflow was correctly used. :heavy_check_mark:
:+1:  Good job!

## STATUS: CHANGES REQUESTED 
_Please Check the comments under the review and make all required changes_
 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

* [OPTIONAL]  Please, add a meaningful  `Project description`. Remember A project description is a high-level overview of why you're doing a project. The document explains a project's objectives and essential qualities. Is an essential document that provides your audience with a comprehensive overview of the project. Think of it as the elevator pitch that focuses on what and why without delving into how. :wink: :100:

* [OPTIONAL ] This is a group project. Please let's include all the authors in the README file. This makes the README file professional and descriptive. 👨🏻‍🎨






 Cheers and Happy coding!clapclapclap computer Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please ping me @Alfredbis29  when you comment so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/classroom_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/classroom_spec.rb]: - Well resolved team 👍🏽

File: spec/student_spec.rb (hunk 2 above)
Comment (Whoistolu) [reply spec/student_spec.rb:21]: - Well resolved team 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Great job implementing the recommended changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**